packages/core/content-manager/server/src/history/controllers/history-version.ts

```diff
@@ -71,14 +71,22 @@ const createHistoryVersionController = ({ strapi }: { strapi: Core.Strapi }) =>
     },
 
     async restoreVersion(ctx) {
-      const versionId: RestoreHistoryVersion.Request['params']['versionId'] = ctx.params.versionId;
+      // @ts-expect-error can't tell koa what I want
+      const request: RestoreHistoryVersion.Request = ctx.request;
```

Comment:
🤷‍♂️ not sure what to do here and I gotta run

Reply:
I believe ctx.request is unknown, you might have to cast the request to match the contract
```suggestion
      const request = ctx.request as RestoreHistoryVersion.Request;
```

Reply:
I had to do one of these guys 😞 
```
const request = ctx.request as unknown as RestoreHistoryVersion.Request;
```

Reply:
For future reference, perhaps not completely worth it here you can also do an assertion function:

```ts
const ctxIsRestoreHistoryVersion = (ctx: unknown): ctx is RestoreHistoryVersion.Request => {
	// ... do stuff
}
```

Obviously the downside is you then have to handle if this assertion does not pass, but i guess in a controller you'd throw a 4XX? 🤷🏼‍♀️

Reply:
agree, we should use a yup validation in the controller (I think zod validations alex was preparing are still not ready)
that could prevent unwanted parameters sent to the body